crates/hir_ty/src/autoderef.rs

```diff
@@ -24,8 +24,10 @@ pub fn autoderef<'a>(
     krate: Option<CrateId>,
     ty: InEnvironment<Canonical<Ty>>,
 ) -> impl Iterator<Item = Canonical<Ty>> + 'a {
+    let _p = profile::span("autoderef");
```

Comment:
Let's remove this one, given that this returns a lazy iterator, this span is confusing